common/djangoapps/student/tests/test_models.py

```diff
@@ -200,6 +205,34 @@ def test_upgrade_deadline_with_schedule_and_professional_mode(self):
         self.assertIsNotNone(enrollment.schedule)
         self.assertIsNone(enrollment.upgrade_deadline)
 
+class PendingNameChangeTests(SharedModuleStoreTestCase):
+    """
+    Tests the deletion of PendingNameChange records
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(PendingNameChangeTests, cls).setUpClass()
+        cls.user = UserFactory()
+        cls.user2 = UserFactory()
+
+    def setUp(self):
+        self.name_change, _ = PendingNameChange.objects.get_or_create(
+            user=self.user,
+            new_name='New Name PII',
+            rationale='for testing!'
+        )
+
+    def test_delete_by_user_removes_pending_name_change(self):
+        self.assertEqual(1, len(PendingNameChange.objects.all()))
+        record_was_deleted = PendingNameChange.delete_by_user_value(self.user, field='user')
+        self.assertTrue(record_was_deleted)
+        self.assertEqual(0, len(PendingNameChange.objects.all()))
+
+    def test_delete_by_user_no_effect_for_user_with_no_name_change(self):
+        self.assertEqual(1, len(PendingNameChange.objects.all()))
+        record_was_deleted = PendingNameChange.delete_by_user_value(self.user2, field='user')
+        self.assertFalse(record_was_deleted)
+        self.assertEqual(1, len(PendingNameChange.objects.all()))
```

Comment:
@bmedx suggested that I check deletions by filtering the model on the user:
`pending_name_changes = PendingNameChange.objects.filter(user=self.user2)`

...then asserting that the search returns nothing:
`self.assertFalse(pending_name_changes.exists())`

He said that `len` on a QuerySet could have performance repercussions in production. Does his comment hold true when checking `len(PendingNameChange.objects.all())`?

Reply:
If it's a unit test with just 1 row in the database. It doesn't matter. That said, doing len() is more expansive than exists() because you would have to wait for all the results to come back before evaluate the size. The exist() function would short circuit at the first encounter.